src/App.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useCallback } from "react";
+import Header from "./components/Header/Header";
+import Sidebar from "./components/Sidebar/Sidebar";
+import { List } from "./components/List/List";
+import { Box } from "@chakra-ui/react";
+import { getCoins } from "./store/coinSlice";
+import { useAppDispatch, useAppSelector } from "./store/store";
+
+export default function App() {
+  const { coins } = useAppSelector(state => state.coin)
+  const dispatch = useAppDispatch();
+  const initApp = useCallback(async () => {
+    await dispatch(getCoins());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setTimeout(() => initApp(), 5000);
+    initApp();
```

Comment:
why two dispatches?

When you use setTimeout in useEffect, set up ref to store timer id & return with clearTimeout function 
[example](https://www.codementor.io/@damianpereira/how-to-use-clearinterval-inside-react-s-useeffect-and-why-it-is-important-1si7mztjlk)